bundle/edu.gemini.qpt.client/src/main/java/edu/gemini/qpt/core/listeners/LimitsListener.java

```diff
@@ -66,9 +82,45 @@ public void propertyChange(PropertyChangeEvent evt) {
 			int percentVisible = (int) (100 * a.getMean(Circumstance.TIMING_WINDOW_OPEN, false));
 			switch (percentVisible) {
 			case 100: // good!
-				if (a.getObs().getTimingWindows().size() > 0)
+				if (a.getObs().getTimingWindows().size() > 0 && night != null)
 					mm.addMarker(true, this, Severity.Info, "Timing constraint is met.", v, a);
-				break;
+
+                    // Report timing windows
+                    final TimingWindowSolver timingWindowSolver = new TimingWindowSolver(a.getObs());
+                    final Iterator<Interval> timingWindowIntervals = timingWindowSolver.solve(night).iterator();
+
+                    if (timingWindowIntervals.hasNext()) {
+                    	StringBuilder sb = new StringBuilder();
+						final Interval timingWindowInterval0 = timingWindowIntervals.next();
+						sb.append(
+							String.format(
+								"Must be in the timing window (%s, %s)",
+								dateFormat.format(Instant.ofEpochMilli(timingWindowInterval0.getStart())),
+								dateFormat.format(Instant.ofEpochMilli(timingWindowInterval0.getEnd()))
```

Comment:
Since we format an `Interval` into a `String` "(t0 - t1)" in several places, it's probably worth extracting that code into a static method.